.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,57 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-18.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+        config-file: ./.github/codeql/codeql-config.yml
+    
+    - name: Install build dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y runc containerd docker.io libmbedtls-dev
+        sudo systemctl start docker
+        sudo systemctl enable docker && sudo chmod 666 /var/run/docker.sock
+        sudo apt-get install python3-setuptools
+
+    - name: Building Mystikos
+      run: |
+        make MYST_PRODUCT_BUILD=1 -j4
```

Comment:
Why do we use PRODUCT_BUILD?

Reply:
I thought this tool scans the source code, not the binary builds?

Reply:
Yes CodeQL scans only the source code. 
I used MYST_PRODUCT_BUILD to reduce build time since it skips building test projects.